src/main/java/org/opensearch/performanceanalyzer/collectors/telemetry/RTFCacheConfigMetricsCollector.java

```diff
@@ -0,0 +1,183 @@
+        super(
+                MetricsConfiguration.CONFIG_MAP.get(RTFCacheConfigMetricsCollector.class)
+                        .samplingInterval,
+                "RTFCacheConfigMetricsCollector",
+                StatMetrics.RTF_CACHE_CONFIG_METRICS_COLLECTOR_EXECUTION_TIME,
+                StatExceptionCode.RTF_CACHE_CONFIG_METRICS_COLLECTOR_ERROR);
+        this.performanceAnalyzerController = performanceAnalyzerController;
+        this.configOverridesWrapper = configOverridesWrapper;
+    }
+
+    @Override
+    public void collectMetrics(long l) {
+        if (performanceAnalyzerController.isCollectorDisabled(
+                configOverridesWrapper, getCollectorName())) {
+            LOG.info("RTFCacheConfigMetricsCollector is disabled. Skipping collection.");
+            return;
+        }
+
+        metricsRegistry = OpenSearchResources.INSTANCE.getMetricsRegistry();
+        if (metricsRegistry == null) {
+            LOG.error("could not get the instance of MetricsRegistry class");
+            return;
+        }
+
+        IndicesService indicesService = OpenSearchResources.INSTANCE.getIndicesService();
+        if (indicesService == null) {
+            LOG.error("could not get the instance of indicesService class");
+            return;
+        }
+
+        LOG.debug("Executing collect metrics for RTFCacheConfigMetricsCollector");
+        CacheMaxSizeStatus fieldDataCacheMaxSizeStatus =
+                AccessController.doPrivileged(
+                        (PrivilegedAction<CacheMaxSizeStatus>)
+                                () -> {
+                                    try {
+                                        Cache fieldDataCache =
+                                                indicesService
+                                                        .getIndicesFieldDataCache()
+                                                        .getCache();
+                                        long fieldDataMaxSize =
+                                                (Long)
+                                                        FieldUtils.readField(
+                                                                fieldDataCache,
+                                                                CACHE_MAX_WEIGHT,
+                                                                true);
+                                        return new CacheMaxSizeStatus(
+                                                FIELD_DATA_CACHE.toString(), fieldDataMaxSize);
+                                    } catch (Exception e) {
+                                        LOG.debug(
+                                                "Error occurred while fetching fieldDataCacheMaxSizeStatus: "
+                                                        + e.getMessage());
+                                        return new CacheMaxSizeStatus(
+                                                FIELD_DATA_CACHE.toString(), -1L);
```

Comment:
What does "-1" mean here. AFAIK -1 is treated as unlimited fieldDataCache?

Reply:
I see there is a check later in the code which says only positive values should be emitted. But then also why do we need -1 handling?